src/erlcloud_application_autoscaler.erl

```diff
@@ -741,8 +741,13 @@ headers(Config, Operation, Body) ->
 %% Extracts and decodes the error from the response returning
 %% in the format of `{error, {ErrorType, Message}}' (matching to how errors are returned in `ercloud_ddb2' module)
 %% Example: {error, {<<"ConcurrentUpdateException">>, <<"You already have a pending update to an Auto Scaling resource.">>}}
-decode_error(Response) ->
-    DecodedError = jsx:decode(Response#aws_request.response_body, [{return_maps, false}]),
-    ErrorType = proplists:get_value(<<"__type">>, DecodedError, <<>>),
-    ErrorMessage = proplists:get_value(<<"Message">>, DecodedError, <<>>),
-    {error, {ErrorType, ErrorMessage}}.
+decode_error(#aws_request{response_body = Body}) ->
+    case jsx:is_json(Body) of
+        false ->
+            {error, {<<>>, <<>>}};
```

Comment:
Here can you actually return `{error, {http_error, Status, StatusLine, RespBody}}`? Should be able to get those all from the `#aws_request`.

Reply:
Updated the PR to return the following:
```erlang
{error, {Response#aws_request.httpc_error_reason, Response#aws_request.response_status, Response#aws_request.response_status_line, Body}};
```